discovery-provider/src/tasks/user_replica_set.py

```diff
@@ -314,9 +329,43 @@ def parse_ursm_cnode_record(update_task, entry, cnode_record):
     # Retrieve endpoint from eth contracts
     cnode_sp_id = event_args._cnodeSpId
     cnode_record.endpoint = get_ursm_cnode_endpoint(update_task, cnode_sp_id)
+
+    if not all_required_fields_present(URSMContentNode, cnode_record):
+        raise EntityMissingRequiredFieldError(
+            "content_node",
+            cnode_record,
+            f"Error parsing content node {cnode_record} with entity missing required field(s)",
+        )
+
     return cnode_record
 
 
+# Update user_record with event arguments
+def parse_user_record(update_task, entry, user_record, block_timestamp):
+    primary = helpers.get_tx_arg(entry, "_primaryId")
+    secondaries = helpers.get_tx_arg(entry, "_secondaryIds")
+    signer = helpers.get_tx_arg(entry, "_signer")
+    user_record.updated_at = datetime.utcfromtimestamp(block_timestamp)
+    user_record.primary_id = primary
+    user_record.secondary_ids = secondaries
+    user_record.replica_set_update_signer = signer
+
+    # Update cnode endpoint string reconstructed from sp ID
+    creator_node_endpoint_str = get_endpoint_string_from_sp_ids(
+        update_task, primary, secondaries
+    )
+    user_record.creator_node_endpoint = creator_node_endpoint_str
+
+    if not all_required_fields_present(User, user_record):
+        raise EntityMissingRequiredFieldError(
+            "user",
+            user_record,
+            f"Error parsing user {user_record} with entity missing required field(s)",
```

Comment:
lets specify user in URSM here to distinguish with regular user

Reply:
good call! done